### PR TITLE
Enhance error handle in bulk load

### PR DESCRIPTION
**Pull request by wayblink**
issue #17777
Signed-off-by: wayblink <anyang.wang@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/import_manager.go [hunk 1]
```diff
@@ -112,9 +112,17 @@ func newImportManager(ctx context.Context, client kv.MetaKv,
...
+		if err != nil {
+			log.Error("importManager init failed, read tasks from Etcd failed, about to panic")
+			panic(err)
+		}
 		// Send out tasks to dataCoord.
-		m.sendOutTasks(ctx)
+		err = m.sendOutTasks(ctx)
+		if err != nil {
+			log.Error("importManager init failed, send out tasks to dataCoord failed, about to panic")
+			panic(err)
```
Comment (yhmo) [inline internal/rootcoord/import_manager.go:124]: sendOutTasks no need to panic.

File: internal/rootcoord/import_manager.go [hunk 2]
```diff
@@ -129,7 +137,11 @@ func (m *importManager) sendOutTasksLoop(wg *sync.WaitGroup) {
 			log.Debug("import manager context done, exit check sendOutTasksLoop")
 			return
 		case <-ticker.C:
-			m.sendOutTasks(m.ctx)
+			err := m.sendOutTasks(m.ctx)
+			if err != nil {
+				log.Error("importManager sendOutTasksLoop fail to send out tasks, about to panic")
+				panic(err)
```
Comment (yhmo) [inline internal/rootcoord/import_manager.go:143]: sendOutTasks no need to panic

File: internal/rootcoord/import_manager.go [hunk 3]
```diff
@@ -205,15 +221,19 @@ func (m *importManager) sendOutTasks(ctx context.Context) error {
...
 
 			log.Debug("import task added as working task", zap.Int64("task ID", it.TaskId))
 			task.State.StateCode = commonpb.ImportState_ImportPending
 			m.workingTasks[task.GetId()] = task
-			m.updateImportTaskStore(task)
+			err := m.updateImportTaskStore(task)
+			return err
 		}()
+		if err != nil {
+			return err
```
Comment (yhmo) [inline internal/rootcoord/import_manager.go:235]: no need return here.
the task has been sent by callImportService, that means a datanode is going to execute the task. the task ought to be put into working list.

File: internal/rootcoord/import_manager.go [hunk 4]
```diff
@@ -626,7 +671,11 @@ func (m *importManager) expireOldTasks(releaseLockFunc func(context.Context, int
 					log.Error("failed to release lock, about to panic!")
 					panic(err)
 				}
-				m.updateImportTaskStore(t)
+				err = m.updateImportTaskStore(t)
+				if err != nil {
+					log.Error("failed to update import task to meta store, about to panic!")
+					panic(err)
```
Comment (yhmo) [inline internal/rootcoord/import_manager.go:677]: I don't think we need panic here.

File: internal/rootcoord/import_manager.go (hunk 1 above)
Comment (soothing-rain) [reply internal/rootcoord/import_manager.go:124]: Agreed. Just logging an error should be good enough here

File: internal/rootcoord/import_manager.go [hunk 5]
```diff
@@ -650,7 +699,11 @@ func (m *importManager) expireOldTasks(releaseLockFunc func(context.Context, int
 					log.Error("failed to release lock, about to panic!")
 					panic(err)
 				}
-				m.updateImportTaskStore(v)
+				err = m.updateImportTaskStore(v)
+				if err != nil {
+					log.Error("failed to update import task to meta store, about to panic!")
+					panic(err)
```
[the remark this answers is not in the document]
Comment (soothing-rain) [reply internal/rootcoord/import_manager.go:705]: Maybe take a look at how other services handle etcd write failure?

File: internal/rootcoord/import_manager.go (hunk 3 above)
Comment (soothing-rain) [reply internal/rootcoord/import_manager.go:235]: We need to first do Etcd update and then put the task in working list. We can then return on Etcd errors.

Comment (wayblink) [reply internal/rootcoord/import_manager.go:235]: like this?
```
		err = func() error {
			m.workingLock.Lock()
			defer m.workingLock.Unlock()

			log.Debug("import task added as working task", zap.Int64("task ID", it.TaskId))
			task.State.StateCode = commonpb.ImportState_ImportPending
			m.workingTasks[task.GetId()] = task
			err := m.updateImportTaskStore(task)
			return err
		}()
		// Erase this task from head of pending list.
		m.pendingTasks = append(m.pendingTasks[:0], m.pendingTasks[1:]...)

		if err != nil {
			return err
		}
```

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Pull request closed without merging**